lib/src/tasks/coverage/api.dart

```diff
@@ -475,4 +631,45 @@ class CoverageTask extends Task {
       return port;
     }
   }
+
+  Future<List<int>> _test_functional(File file) async {
+
+    int count = observatoryPort.length;
+
+    String executable = 'pub';
+    List args = ['run',
+//        int port = await getOpenPort();
+//        String executable = 'dart';
+//        List args = ["--observe=${port}"
+    file.path.split(config.test.functionalTests[0] + '/')[1]];
+    print(config.test.functionalTests[0]);
+    TaskProcess process = new TaskProcess(executable, args,workingDirectory:config.test.functionalTests[0]);
```

Comment:
Is the only difference between running a functional test and a regular test that the functional test is run as a child process via `pub run`? Could every test be run like this?